hvac/v1/__init__.py

```diff
@@ -1244,10 +1247,19 @@ def _put(self, url, **kwargs):
     def _delete(self, url, **kwargs):
         return self.__request('delete', url, **kwargs)
 
-    def __request(self, method, url, headers=None, **kwargs):
+    def __request(self, method, url, **kwargs):
+        # NOTE(rswedenburg): added ability to retain the vault url's path when it's needed (for example when referencing a service path in Mesosphere)
+        if self.keep_url_path:
+            split_url = urlsplit(self._url)
+            url = split_url[2] + url
         url = urljoin(self._url, url)
 
-        if not headers:
+        # NOTE(rswedenburg): assuming that the user wants to retain the headers on the session passed in, this grabs and uses them
+        if self.session:
+            headers = self.session.headers
+            if not headers:
+                headers = {}
```

Comment:
This breaks some previous usage of this method. If the method had custom headers passed in previously they no longer would be referenced here.